keps/sig-api-machinery/4222-cbor-serializer/README.md

```diff
@@ -0,0 +1,1149 @@
+
+
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+Tests for the following behaviors will be added:
+
+- decoding a map containing duplicate keys into a Go map produces an error
+- decoding a map containing duplicate keys into a Go struct produces an error
+- roundtripping preserves the distinction between integers and floating-point
+  numbers
+- decoding a text string containing an invalid UTF-8 sequence produces an error
+- decoding a map into a Go struct matches `json` field tag names
+  case-sensitively
+- when decoding a map into a Go struct, a case-insensitive match between a key
+  and a `json` field tag name is treated the same as no match
+- encoding a struct with duplicate field names (`json` tag names) does not
+  result in a map containing duplicate keys ([https://go.dev/issue/17913])
+- pooled buffers should not grow and be retained forever
+  ([https://go.dev/issue/23199])
+- decoding into a Go interface{} stores only either `nil` concrete values or
+  concrete values of type `bool`, `string`, `int64`, `float64`, `[]interface{}`,
+  or `map[string]interface{}` (no special treatment of tagged content producing
+  time.Time, math/big.Int, etc.)
+- conformance to CBOR specification (adopt existing suite and/or develop as necessary)
```

Comment:
Being able to test the CBOR implementations of multiple client languages (Go, Java, Python...) would be great here

Reply:
Definitely. Updated to make this a requirement for conformance testing.